src/test/java/com/kodilla/ecommercee/domain/ProductTestSuite.java

```diff
@@ -0,0 +1,165 @@
+        Assert.assertTrue(loadedProduct.isPresent());
+        Assert.assertEquals("Shirt", loadedProduct.get().getName());
+        Assert.assertEquals(BigDecimal.valueOf(5.20), loadedProduct.get().getPrice());
+
+    }
+
+    @Transactional
+    @Test
+    public void testRelationWithGroups() {
+        //given
+        Group clothes = new Group();
+        Group food = new Group();
+
+        groupRepository.save(clothes);
+        groupRepository.save(food);
+
+        Product trousers = new Product("trousers", "normal trousers", BigDecimal.valueOf(22.90));
+        trousers.setGroup(clothes);
+
+        Product tomato = new Product("tomato", "red tomato", BigDecimal.valueOf(0.99));
+        tomato.setGroup(food);
+
+        productRepository.save(trousers);
+        productRepository.save(tomato);
+
+        Long clothesId = clothes.getId();
+        Long foodId = food.getId();
+
+        Long trousersId = trousers.getId();
+        Long tomatoId = tomato.getId();
+
+        //when
+        Product loadedTrousers = productRepository.findById(trousersId).get();
+        Long loadedTrousersGroupId = loadedTrousers.getGroup().getId();
+
+        Product loadedTomato = productRepository.findById(tomatoId).get();
+        Long loadedTomatoGroupId =loadedTomato.getGroup().getId();
+
+        //then
+        Assert.assertEquals(clothesId, loadedTrousersGroupId);
+        Assert.assertEquals(foodId, loadedTomatoGroupId);
+    }
+
+    @Transactional
+    @Test
+    public void testFindAll() {
+        //given
+        Product testProductOne = new Product("Shirt", "white shirt", BigDecimal.valueOf(5.20));
+        Product testProductTwo = new Product("Hat", "fancy hat", BigDecimal.valueOf(8.99));
+        Product testProductThree = new Product("Pants", "panties", BigDecimal.valueOf(3.00));
+        Product testProductFour = new Product("Scarf", "nice scarf", BigDecimal.valueOf(7.05));
+
+        productRepository.save(testProductOne);
+        productRepository.save(testProductTwo);
+        productRepository.save(testProductThree);
+        productRepository.save(testProductFour);
+
+        //when
+        List<Product> loadedProducts = productRepository.findAll();
+
+        //then
+        Assert.assertTrue(loadedProducts.contains(testProductOne));
+        Assert.assertTrue(loadedProducts.contains(testProductTwo));
+        Assert.assertTrue(loadedProducts.contains(testProductThree));
+        Assert.assertTrue(loadedProducts.contains(testProductFour));
+    }
+
+    @Test
+    public void testDeleteProduct() {
+        //given
+        Product testProduct = new Product("Shirt", "white shirt", BigDecimal.valueOf(5.20));
+        productRepository.save(testProduct);
+
+        //when
+        productRepository.delete(testProduct);
+        List<Product> allProducts = productRepository.findAll();
+
+        //then
+        Assert.assertTrue( !allProducts.contains(testProduct) );
+
+    }
+
+    @Test
+    public void testDeleteProductById() {
```

Comment:
Na początku napisałem zupełnie inny komentarz ale potem sprawdziłem i opatrzenie tego testu @transactional niczego nie psuje:

2019-02-14 00:14:36.470 INFO 19977 --- [ main] o.s.t.c.transaction.TransactionContext : Began transaction (1) for test context [DefaultTestContext@4135c3b testClass = ProductTestSuite, testInstance = com.kodilla.ecommercee.domain.ProductTestSuite@7a2a7492, testMethod = testDeleteProduct@ProductTestSuite, testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@6302bbb1 testClass = ProductTestSuite, locations = '{}', classes = '{class com.kodilla.ecommercee.EcommerceeApplication}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}', contextCustomizers = set[org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@52feb982, org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@bd8db5a, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@4b553d26, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@2c039ac6], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> true, 'org.springframework.test.context.web.ServletTestExecutionListener.populatedRequestContextHolder' -> true, 'org.springframework.test.context.web.ServletTestExecutionListener.resetRequestContextHolder' -> true]]; transaction manager [org.springframework.orm.jpa.JpaTransactionManager@1f45db49]; rollback [true]
Hibernate: select next_val as id_val from hibernate_sequence for update
Hibernate: update hibernate_sequence set next_val= ? where next_val=?
2019-02-14 00:14:36.820 INFO 19977 --- [ main] o.h.h.i.QueryTranslatorFactoryInitiator : HHH000397: Using ASTQueryTranslatorFactory
Hibernate: insert into products (description, group_id, name, price, product_id) values (?, ?, ?, ?, ?)
Hibernate: delete from products where product_id=?
Hibernate: select product0_.product_id as product_1_4_, product0_.description as descript2_4_, product0_.group_id as group_id5_4_, product0_.name as name3_4_, product0_.price as price4_4_ from products product0_
2019-02-14 00:14:37.808 INFO 19977 --- [ main] o.s.t.c.transaction.TransactionContext : Rolled back transaction for test: